src/deadline/cinema4d_submitter/scene.py

```diff
@@ -170,6 +174,40 @@ def get_render_data(doc=None, take=None):
             render_data = doc.GetActiveRenderData()
         return render_data
 
+    @staticmethod
+    def get_output_paths(take=None) -> str:
```

Comment:
Hmm... If I do a diff between a methods, that difference in functionality doesn't exist? It looks like one is missing pulling the default take data, we may want to add that? `get_output_paths` appears to otherwise return the exact same data as `get_output_directories`, just formatted differently (ie. dirname for asset references and in a set). Which if we're adding them as PATH + OUTPUT job parameters to the template will automatically handle that. 

If there is an actual difference between the two, then I ask that we document what that is, otherwise I'd err on having the one function and if we're still manually adding the asset reference for the outputs, then we can take the dirname of the result of one function. 

```diff
-    def get_output_directories(render_data=None, take=None) -> set[str]:
-        """
-        Returns a list of directories files will be output to.
-        """
+    def get_output_paths(take=None) -> str:
         doc = c4d.documents.GetActiveDocument()
         doc_path = doc.GetDocumentPath()
-        if not take:
-            take_data = doc.GetTakeData()
-            take = take_data.GetCurrentTake()
+
+
+        
         render_data = Scene.get_render_data(doc=doc, take=take)
-        rbc = render_data.GetDataInstance()
+        render_base_container_instance = render_data.GetDataInstance()
         rpd = {
             "_doc": doc,
             "_rData": render_data,
-            "_rBc": rbc,
+            "_rBc": render_base_container_instance,
             "_frame": doc.GetTime().GetFrame(doc.GetFps()),
         }
         if take:
             rpd["_take"] = take
-        image_paths = set()
+        default_out = ""
+        multi_out = ""
         if render_data[c4d.RDATA_SAVEIMAGE]:
             path = render_data[c4d.RDATA_PATH]
             xpath = c4d.modules.tokensystem.FilenameConvertTokens(path, rpd)
@@ -25,7 +23,7 @@
                 if xpath.startswith("./"):
                     xpath = xpath[2:]
                 xpath = os.path.join(doc_path, xpath)
-            image_paths.add(os.path.dirname(os.path.normpath(xpath)))
+            default_out = os.path.normpath(xpath)
         if render_data[c4d.RDATA_MULTIPASS_SAVEIMAGE]:
             path = render_data[c4d.RDATA_MULTIPASS_FILENAME]
             xpath = c4d.modules.tokensystem.FilenameConvertTokens(path, rpd)
@@ -33,5 +31,5 @@
                 if xpath.startswith("./"):
                     xpath = xpath[2:]
                 xpath = os.path.join(doc_path, xpath)
-            image_paths.add(os.path.dirname(os.path.normpath(xpath)))
-        return image_paths
\ No newline at end of file
+            multi_out = os.path.normpath(xpath)
+        return default_out, multi_out
\ No newline at end of file

```

Reply:
howdy @epmog it's common in other submitters to have 2 methods for this - for example in maya [get_output_directories()](https://github.com/aws-deadline/deadline-cloud-for-maya/blob/mainline/src/deadline/maya_submitter/scene.py#L119) and [output_path()](https://github.com/aws-deadline/deadline-cloud-for-maya/blob/mainline/src/deadline/maya_submitter/scene.py#L136) are the equivalent of our get_output_directories() and get_output_paths() respectively -- the only difference is we return a tuple default_out, mulit_out here. It's a feature that get_output_paths() doesn't fall back to main take, if it did the output mapping of each take would be overridden with the main take path - this is what I mean about 2 use cases needing 2 different functions, combining them can open the door to side effects that break other features.  I'll add a doc string to get_output_paths() to make this clearer.